api.yml

```diff
@@ -19,9 +19,97 @@ paths:
     post:
       tags:
         - internal
-      summary: Verifies a credential and returns its verification status.
+      summary: Verify a given credential.
       operationId: verifyCredential
-      description: Verifies a credential and returns its verification status. Support of this part of the API is REQUIRED for implementations.
+      description: Performs a series of verification checks on a provided Verifiable Credential and returns the overall status of the checks as well as the status of each individual check. Support of this part of the API is REQUIRED for implementations.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              description: The credential to be verified and a set of options to use when verifying the credential.
+              type: object
+              properties:
+                verifiableCredential:
+                  type: object
+                  additionalProperties: false
+                  description: The verifiable credential to check.
+                  example: {
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1"
+  ],
+  "id": "http://example.edu/credentials/3732",
+  "type": [
+    "VerifiableCredential",
+    "UniversityDegreeCredential"
+  ],
+  "issuer": "did:example:c6f1c276e12ec21ebfeb1f712eb",
+  "issuanceDate": "2020-03-11",
+  "credentialSubject": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "degree": {
+      "type": "BachelorDegree",
+      "name": "Bachelor of Science and Arts"
+    }
+  },
+  "proof": [{
+    "type": "Ed25519Signature2018",
+    "created": "2020-03-11T23:09:06.803Z",
+    "jws": "eyJhbGciOiJFZERTQSIsImI2..hS5SPmWTFocjcDg",
+    "proofPurpose": "assertionMethod",
+    "verificationMethod": "did:example:c6f1c276e12ec21ebfeb1f712eb#jf893k"
+  }]
+}
+                options:
+                  type: object
+                  additionalProperties: false
+                  description: Options for specifying how the credential will be issued.
+                  properties:
+                    checks:
+                      type: array
+                      items:
+                        type: string
+                      description: An array of verification checks to be performed on the credential. WARNING - Implementers are cautioned that the list of checks is currently incomplete and other checks such as issuance, expiration, nonce, domain, and acceptable issuers are expected to be added in time. A default list is also expected to encapsulate all available checks. This option might be changed to a 'ignore' option where it lists the checks the developer would like to skip.
```

Comment:
Where will these string values be defined? I don't think we need them defined right now to make progress on the general shape of the API -- but I think they should be defined by the time we get to 1.0.

Reply:
> Where will these string values be defined?

My expectation is that they'll start out being defined in the spec and eventually there will be a registry so that people can add their own checks.

This api.yml file is only going to get us so far... I have plans to convert/absorb all of this in ReSpec format so we can add many of the warnings and concerns that you and others have been outlining throughout the fevered process of putting this API together (e.g., this API is meant to be used within the same identity boundary for the issuer, other APIs must be used to communicate with people outside of the identity boundary, etc.)